### PR TITLE
adding support for augio

### PR DESCRIPTION
**Pull request by danforthewin**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/augio/Augio.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#ifndef AUGIO_H
+#define AUGIO_H
```
Comment (drashna) [inline keyboards/augio/Augio.h:2]:
```suggestion
#pragma once
```

File: keyboards/augio/Augio.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	{ K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17 }, \
+	{ K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27 }, \
+	{ K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37 }, \
+	{ K40,   K41,   K42,   K43,   K44,   K45,   K46,   K47 }, \
+	{ K50,   K51,   K52,   K53,   K54,   K55,   K56,   K57 }, \
+	{ K60,   K61,   K62,   K63,   KC_NO, K65,   K66,   K67 }, \
+	{ K70,   K71,   K72,   K73,   K74,   K75,   K76,   K77 }  \
+}
+
+#endif
```
Comment (drashna) [inline keyboards/augio/Augio.h:22]:
```suggestion
```

File: keyboards/augio/config.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (drashna) [inline keyboards/augio/config.h:2]:
```suggestion
#pragma once
```

File: keyboards/augio/config.h [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    qmkbuilder
+#define PRODUCT         keyboard
+#define DESCRIPTION     Keyboard
```
Comment (drashna) [inline keyboards/augio/config.h:12]: You shoudl fill in the manufacturer
```suggestion
#define MANUFACTURER    ???
#define PRODUCT         Augio
```

File: keyboards/augio/config.h [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/augio/config.h:33]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/augio/config.h [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/augio/config.h:47]: Not needed. These are default now, and should be removed. 

```suggestion
```

File: keyboards/augio/config.h [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 0
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
+
+#endif
```
Comment (drashna) [inline keyboards/augio/config.h:58]:
```suggestion
```

File: keyboards/augio/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+void matrix_scan_user(void) {
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	return true;
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (drashna) [inline keyboards/augio/keymaps/default/keymap.c:96]: None of these are used or have been deprecated, and should be removed. 

```suggestion
```

File: keyboards/augio/rules.mk [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/augio/rules.mk:41]: This is all set by default, now.

```suggestion
```

File: keyboards/augio/rules.mk [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
```
Comment (drashna) [inline keyboards/augio/rules.mk:57]:
```suggestion
BOOTLOADER = caterina


BOOTMAGIC_ENABLE = lite  # Virtual DIP switch configuration
MOUSEKEY_ENABLE  = yes   # Mouse keys
EXTRAKEY_ENABLE  = yes   # Audio control and System control
CONSOLE_ENABLE   = no    # Console for debug
COMMAND_ENABLE   = no    # Commands for debug and configuration
SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
NKRO_ENABLE      = yes   # USB Nkey Rollover
BACKLIGHT_ENABLE = no    # Enable keyboard backlight functionality
AUDIO_ENABLE     = no
RGBLIGHT_ENABLE  = no
```

**Comment by drashna** [comment]
Just a heads up, it looks like you used kbfirmware to generate the firmware here.  It is ... very much out of date (3+ years now), and there are a number of things that we'd like to see updated to be in line with current standards.

**Comment by danforthewin** [comment]
Thanks for the heads up. Yeah, I used kbfirmware to help me get it started.
I'll make those changes you suggested and get back to you.

Thanks!

File: keyboards/augio/config.h [hunk 11]
```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
```
Comment (Erovia) [inline keyboards/augio/config.h:6]:
```suggestion
#define VENDOR_ID       0x6477  // DW 
```
Just to make it unique.

File: keyboards/augio/keymaps/default/keymap.c (hunk 8 above)
Comment (Erovia) [reply keyboards/augio/keymaps/default/keymap.c:96]: They are still here.

File: keyboards/augio/config.h [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
```
Comment (fauxpark) [inline keyboards/augio/config.h:29]:
```suggestion
```

File: keyboards/augio/rules.mk [hunk 13]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = atmega32u4
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTLOADER = caterina
+
```
Comment (fauxpark) [inline keyboards/augio/rules.mk:8]:
```suggestion
# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   comment out to disable the options.
#
```

File: keyboards/augio/Augio.c [hunk 14]
```diff
@@ -0,0 +1 @@
+#include "augio.h"
```
Comment (fauxpark) [inline keyboards/augio/Augio.c:1]: Please rename this file to lowercase.

File: keyboards/augio/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,73 @@
+#include "augio.h"
```
Comment (drashna) [inline keyboards/augio/keymaps/default/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/augio/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+
+enum augio_layers {
+  _QWERTY,
+  _LOWER,
+  _RAISE,
+  _ADJUST
```
Comment (drashna) [inline keyboards/augio/keymaps/default/keymap.c:8]:
```suggestion
```

File: keyboards/augio/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+enum augio_keycodes {
+  QWERTY = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  ADJUST
+};
```
Comment (drashna) [inline keyboards/augio/keymaps/default/keymap.c:16]: This isn't needed, as you're not using it.
```suggestion
```

File: keyboards/augio/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+	 * |------+------+------+------++------+------+------+------+------+------+------+------+------+------+------+------|
+	 * |      |      |      |      ||      |      |      |      |      |             |      |      |      |      |      |
+	 * `----------------------------------------------------------------------------------------------------------------'
+	 */
+	[_ADJUST] = LAYOUT(
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+		_______, _______, _______, _______, _______, _______, _______, _______, _______,     _______, 	   _______, _______, _______, _______, _______
+	)
```
Comment (drashna) [inline keyboards/augio/keymaps/default/keymap.c:72]:
```suggestion
```

**Pull request closed without merging**